addons/sys_core/fnc_hasItem.sqf

```diff
@@ -0,0 +1,21 @@
+/*
+ * Author: ACRE2Team
+ * Checks if a unit has a specific item.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Item to be checked <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, "acre2_vhf30108"] call acre_sys_core_fnc_hasItem
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params [["_unit", objNull], ["_item", ""]];
+
+_item in (items _unit)
```

Comment:
This is pretty slow, might be worth looking at https://github.com/acemod/ACE3/pull/6350 (first point of caching items and using `getItemCargo`).